autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java

```diff
@@ -104,10 +108,11 @@ public void run() {
 
     private void scaleUp(int scalingAdjustment, int totalNodes) {
         metricService.incrementMetricCounter(MetricType.CLUSTER_UPSCALE_TRIGGERED);
-        if (scalingHardLimitsService.isViolatingMaxUpscaleStepInNodeCount(scalingAdjustment)) {
-            LOGGER.debug("Upscale requested for '{}' nodes. Upscaling with the maximum allowed of '{}' node(s)",
-                    scalingAdjustment, scalingHardLimitsService.getMaxUpscaleStepInNodeCount());
-            scalingAdjustment = scalingHardLimitsService.getMaxUpscaleStepInNodeCount();
+        int maxScaleUpStepAllowed = entitlementValidationService.scalingStepEntitlementEnabled(cluster.getClusterPertain().getTenant()) ?
```

Comment:
Here, and in Periscope ... can this be changed to test against the default first. If that fails, check with the entitlement service. (Avoid making calls to the entitlement service for every such request)